x-pack/plugins/search_indices/public/utils/indices.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// see https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html for the current rules
```

Comment:
question: we have a few of these index validation functions. Is this copied from somewhere?

Reply:
It's in at least 4 places, so we should look at moving it to a package, especially since I just noticed some of them have an extra check for `.` & `..`

Reply:
yeh +1 on separate package, especially between index_management and us.